policy/scan/local_scanner.go

```diff
@@ -334,107 +420,50 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 
 	log.Info().Msgf("discover related assets for %d asset(s)", len(job.Inventory.Spec.Assets))
 
-	var assets []*assetWithRuntime
-	assetList, assetCandidates, assetErrors, err := createAssetCandidateList(ctx, job, upstream, s.recording)
+	discoveredAssets, err := discoverAssets(ctx, job, upstream, s.recording)
 	if err != nil {
 		return nil, err
 	}
 
 	// if we had asset errors we want to place them into the reporter
-	for i := range assetErrors {
-		reporter.AddScanError(assetErrors[i].asset, assetErrors[i].err)
+	for i := range discoveredAssets.errors {
+		reporter.AddScanError(discoveredAssets.errors[i].asset, discoveredAssets.errors[i].err)
 	}
 
+	// TODO: call the function below with the --platform-id CLI argument (if it is set)
```

Comment:
this never worked in the latest cnspec versions. Need to figure out how to get the value of the `--platform-id` flag, to do the filtering